### PR TITLE
Allow configuring `rustdoc --disable-minification` in config.toml

### PR DESCRIPTION
**Pull request by notriddle**
This way, you can debug rustdoc's JavaScript and CSS file with normal F12 Dev Tools and you'll have useful line numbers to work with.

### CODE REVIEW

**Comment by jyn514** [comment]
Would it maybe be more useful to include a source map instead?

**Comment by jyn514** [comment]
Oh wait, this is changing bootstrap, not rustdoc. Why does rustdoc even have a `--disable-minification` option??

**Comment by jyn514** [comment]
It looks like it's still unstable - what do you think about removing it in favor of unconditionally minifying and generating a source map?

cc @GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
I use `--disable-minification` often. However, I'm not sure if using source maps would be very useful in our case considering it's only minification.

**Comment by jyn514** [comment]
> I use --disable-minification often. However, I'm not sure if using source maps would be very useful in our case considering it's only minification.

I don't understand - if you use disable-minification often, why wouldn't you find source maps useful? They do the same thing, but source maps can be included unconditionally instead of having to change the build process.

**Comment by GuillaumeGomez** [comment]
Hum, that's actually a good point. Let's go for source maps then!

**Comment by notriddle** [comment]
The big downside is that source map generation is something you need to implement, and the `minifier` crate doesn't. It's a complicated feature that will take a few person-days to get it good enough to use. And when a minifier has a bug, it can take further hours of diving before you even realize that it's a bug in the minifier and not a bug in the actual main.js code, and fixing them can get pretty complex, because these can be large and complex beasts.

Adding an option to turn off minification took a few minutes, and can be merged right now, and if it has a bug, it'll fail by minifying or not minifying the code when you expect, which will be obvious.

**Comment by jyn514** [comment]
Given that this already exists, I'm ok with adding an option for this to bootstrap (although @Mark-Simulacrum may want you to just use `RUSTDOCFLAGS=--disable-minification` instead). But I think we shouldn't plan to stabilize this option, and see if we can add source maps to make up for it.

**Comment by Mark-Simulacrum** [comment]
I think adding an option like this PR does is fine. I think it's reasonable to expose developer-only options via config.toml.

@bors r+ rollup

File: src/bootstrap/config.rs [hunk 1]
```diff
@@ -663,6 +665,7 @@ impl Config {
         config.python = build.python.map(PathBuf::from);
         set(&mut config.low_priority, build.low_priority);
         set(&mut config.compiler_docs, build.compiler_docs);
+        set(&mut config.docs_minification, build.docs_minification);
```
Comment (jyn514) [inline src/bootstrap/config.rs:668]: Are you sure this gets set to `true` if the option isn't set?

Comment (Mark-Simulacrum) [reply src/bootstrap/config.rs:668]: Oh, good catch. Yeah, seems like that can be adjusted.

**Pull request merged**